README.md

```diff
@@ -311,3 +312,38 @@ kept the flags field for debug. In hind sight, it would have made more sense to
 use a separate header such as `X-B3-Debug: 1` as no other flags were added and
 no implementation treats `X-B3-Flags` as a bit field. That said, renaming the
 field would cause more damage than good, so it was left alone.
+
+## What is the relationship between B3 context and trace data formats?
+
+When you see B3 headers of any kind, you cannot assume the out-of-band data is
+in Zipkin format, or even that data is sent to a tracing system. For example,
+some embed trace data into log lines correlated with the same IDs in the headers.
+Others use B3 headers for dramatically different trace formats.
+
+That said, it is typically the case that data is sent out-of-band, and outgoing
+fields match some fields in that data. For example, in [Zipkin format](https://github.com/openzipkin/zipkin-api), all fields
+except Accept and Deny flags are reported. That's because Accept is implied and
+Deny would not result in reported data.
+
+### Handling B3 when reporting data to a tracing system
+
+It is possible that a proxy, such as Envoy, accepts trace data correlated with B3
+headers sent by the application. Depending on configuration, that proxy could
+start a trace for the act of sending trace data, amplifying traffic to your
+tracing system with low or negative value traces. While most sites will not trace
+data sent to a tracing system, there are generally 2 ways to address this proxy
+setup:
+
+1. Configure the proxy to not trace your trace reporting endpont (Ex in Zipkin, `POST /api/v2/spans`)
+2. Send `b3: 0` header (an explicit Deny), to prevent tracing.
+
+We recommend having your trace reporting library send `b3: 0`. This cements a
+cost to everyone, only to help those proxying their trace data. However, it is
+becoming common in mesh environments to proxy everything, and in reality there is
+limited ability for end users to change the sampling policy of proxies.
```

Comment:
Is https://docs.aws.amazon.com/xray/latest/devguide/xray-sdk-go-configuration.html#xray-sdk-go-configuration-sampling is a sample of "sampling policy" in this context?

Reply:
exactly.

spring-cloud-sleuth has similar except the policy can merge from different sources, ex a deny list for well known paths that shouldn't be traced, and user-supplied policy.

however you compose the list, it would look like amazon's format in abstract, especially subject to a request rate (per second) is better than a percentage (0.1). amazon I think mixes these words a little but in general the format is good.

Reply:
an interesting more comprehensive format for trace config is this, mentioning for research, not necessarily direct input to how proxy config works https://github.com/kamon-io/Kamon/blob/master/core/kamon-core/src/main/resources/reference.conf#L218-L226

hats off to @ivantopo  and @dpsoft for the design